v0/destinations/am/transform.js

```diff
@@ -201,6 +226,21 @@ function responseBuilderSimple(
   let endpoint = ENDPOINT;
   let traits;
 
+  if (EventType.IDENTIFY) {
+    // If mapped to destination, Add externalId to traits
+    if (get(message, MappedToDestinationKey)) {
+      addExternalIdToTraits(message);
+      const identifierType = get(
+        message,
+        "context.externalId.0.identifierType"
+      );
+      if (identifierType === "user_id") {
```

Comment:
why can't we run this for "device_id"?

Reply:
We do not need to map deviceId with rudder's userId right?

Reply:
I see.
So, u are saying we treat `deviceId` and `userId` differently while transforming (in this file)?
Then do we need to set `deviceId` just like we are setting `userId`?

Reply:
we can assign device_id/user_id for am. 
We do always have userId in our payload.
Only when user is assigning user_id as an identifier then only we replace the userId in the payload.